### DISCUSSION TITLE
ボールと一列のキーが反応したりしなかったりします

### DISCUSSION THREAD

**Discussion by Y-KBYM** [Yowkees/keyball#347] [Q&A]
keyball39、右ボール仕様で製作を進めています。
FWは、keyball39_viaを導入しており、ヨーキースさんのビルドログに沿ってきました。
途中のFW書き込みで正規FWを書き込んで、全てのキーの導通は確認済みでした。

全て組み終わって動かしたところ、トラックボールと右の上から2列目が反応しませんでした。キーボードは反応して問題ない時もありましたが、何度か試すうちにまったく反応がなくなりました。トラックボールはまったく反応ありません。

キーボードの反応について過去ログを見ましたところ、ProMicroのハンダが原因らしいとのことでやり直しをしました。その際、ピンヘッダとProMicroの上に出ているピンとの間で導通は確認して問題ありませんでした。

トラックボールについては、#51、#274、#276でやり取りされていた確認内容は全て確認し、ジャンパなどのハンダ、センサとProMicroの導通、ProMicro入替を実施しましたが、特に問題はありませんでした。

その他確認すべきところや、テスターをあたるところなどアドバイスいただけましたら幸甚です。ハンダの様子をアップいたしますので、あわせてご確認ください。

追伸：再度FWを書き込もうとしたところ、Reading bytes was timeoutというエラーが出たのですが、これは問題ないのでしょうか？組み立て中に書き込んだ時にはこのようなエラーは出てなかったと思うのですが・・・。
このエラーが出てから後にRemapでキーアサインを変更してみましたが、特に問題なく変更できました。

![134020](https://github.com/Yowkees/keyball/assets/145252214/1a325605-b510-4f29-9575-2b84c1a1ffc0)
![134022_0](https://github.com/Yowkees/keyball/assets/145252214/30be0894-9270-47ee-9c94-14d5423cd4e7)
![134023](https://github.com/Yowkees/keyball/assets/145252214/1996c5e1-0efc-4607-90f7-35711bee9c7c)
![remap-test](https://github.com/Yowkees/keyball/assets/145252214/30bb4c86-3545-479c-9aa3-db1df6cf2ac8)
![ファームウェア書き込み不良](https://github.com/Yowkees/keyball/assets/145252214/46080d3c-4524-4277-bc43-e5008a2a4f83)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

お写真を拝見した所、半田付けには特に不具合となりそうな個所はありませんでした。
ファームウェア書き込み時ですが、初回以外は書き込み直前にリセットスイッチをすばやく２回押してProMicroの動作を停止する必要がありますので、お願いします。

２点ご確認ください。
①トラックボールについて
お手数ですがトラックボール基板のお写真も見せていただく事はできますか？また、念のためProMicro本体の半田付けのお写真もお願いします。
その際、トラックボール基板を取り外しますが、ここの7Pコンスルーの抜き差しでも復活することがありますので、ご確認ください。

②キー1列の不具合について、
不具合のあるキーのPCBソケットですが、ちょっと半田の量が多いようです。こちらを半田吸い取り線等で除去して少なくできますでしょうか。あまり半田が多いと形状が膨らみますので、隣のGND等のパターンにショートすることがあります。

以上、お手数ですがご確認、ご対応お願いいたします。

**Reply by Y-KBYM** [reply]
Yowkees様
コメントありがとうございます。

ProMicroの入れ替えを再度試そうとして、ProMicroを抜いたところ、トラックボール側のピンの一部がほんの少し（1mmもないぐらい）浮いているのが見つかりました。ピンソケット内部にハンダが少し上がってきているのが見えましたので、コンスルーに付け替えたところトラックボールも反応し、キー入力も問題なくできました。
お騒がせしました。ありがとうございました。

トラックボールが動くようになって、感度の変更について確認したいことがございますので、別途アップいたします。

**Comment by Yowkees** [comment]
ご自身でコンスルーの確認とご解決いただき、誠にありがとうございました。
今後ともどうぞよろしくお願いいたします。

**Reply by Y-KBYM** [reply]
Yowkees様
ご対応ありがとうございました。
最初にもっとしっかり確認しておけば、もっと早く解決できたのですが、何とか解決できました。
また何かありましたら、よろしくお願いします。
